### DISCUSSION TITLE
Support for plural and alternate names for ingredients/tools/units

### DISCUSSION THREAD

**Discussion by terrapinlogo** [mealie-recipes/mealie#1219] [Feature Request]
**Before submitting this feature request I have**
- [x] Reviewed the documentation and verified that my feature request isn't already a feature.
- [x] Reviewed the existing feature requests and my feature has not been requested.
- [x] Checked the [tasks taged](https://github.com/hay-kot/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

**Please Describe The Problem To Be Solved**

Support alternate names for items with a special case for the item plural that is used when qty is > 1.
 
eg.
ingredient name: coriander
alternate name: cilantro or vise versa depending on user preference

ingredient name: Spring Onion       - Users preferred name
plural: Spring Onions                       - Users preferred plural
Alternate Names: scallion, scallions, green onion, green onions        -alternate names and spellings, including plurals

Any time the user scrapes, parses or searches for a plural or alternate name, return the preferred name.

**(Optional): Suggest A Solution**

Another two fields in the DB for each ingredient/tool/unit, one for the plural (can be the same as the preferred name), and a list of alternates including their plurals. These would then be used as extra keys to return the preferred name and id

**Additional Information**
- [ ] If the feature is accepted I'm willing to submit a PR to provide this feature

![image](https://user-images.githubusercontent.com/5179945/168412841-a3ab355e-5819-4a42-a60d-f47eb9c0dc71.png)

**Comment by terrapinlogo** [comment]
This could also be used for different languages

**Comment by hay-kot** [comment]
Tracking 

- https://github.com/hay-kot/mealie/issues/1242
